### PR TITLE
Difficulty adjustment each block based on the average block time for …

### PR DESCRIPTION
**Pull request by martin-key**
…the last 10 blocks.

### CODE REVIEW

File: src/pow.cpp [hunk 1]
```diff
@@ -14,18 +14,14 @@
 #include "uint256.h"
 #include "util.h"
 
+#define NUM_OF_BLOCKS_DIFF_ADJ 10
```
Comment (h4x3rotab) [inline src/pow.cpp]: I'm afraid 10 blocks is not enough to prevent volatile difficulty fluctuation. But it can be discussed

File: src/pow.cpp [hunk 2]
```diff
@@ -14,18 +14,14 @@
...
     unsigned int nProofOfWorkLimit = UintToArith256(params.powLimit).GetCompact();
 
     int nHeightNext = pindexLast->nHeight + 1;
-    if (nHeightNext >= params.BTGHeight  && nHeightNext < params.BTGHeight + params.BTGPremineWindow)
-    {
-        // Lowest difficulty for Bitcoin GPU premining period.
-        return nProofOfWorkLimit;
-    }
-    else if (nHeightNext % params.DifficultyAdjustmentInterval() != 0)
+ if (nHeightNext % params.DifficultyAdjustmentInterval() != 0)
```
Comment (h4x3rotab) [inline src/pow.cpp]: Do you want to calculate the difficulty by a sliding window?

File: src/pow.cpp [hunk 3]
```diff
@@ -47,8 +43,8 @@ unsigned int GetNextWorkRequired(const CBlockIndex* pindexLast, const CBlockHead
         return pindexLast->nBits;
     }
 
-    // Go back by what we want to be 14 days worth of blocks
-    int nHeightFirst = pindexLast->nHeight - (params.DifficultyAdjustmentInterval()-1);
+    // Calculate the diff for the next block based on the block time in the last 10 blocks
+    int nHeightFirst = pindexLast->nHeight - NUM_OF_BLOCKS_DIFF_ADJ;
```
Comment (h4x3rotab) [inline src/pow.cpp]: Maybe nHeightFirst = min(0, nHeightFirst)?
Also, I believe the new consensus rule should be only applied to blocks after the fork.

File: src/pow.cpp (hunk 2 above)
Comment (StarbuckBG) [reply src/pow.cpp]: No, I have added a new params for the blockchain consensus that is calculating the difficulty, each block.

**Pull request closed without merging**